src/main/java/net/wasdev/wlp/common/plugins/util/InstallFeatureUtil.java

```diff
@@ -0,0 +1,784 @@
+            return getServerFeatures(serverDirectory);
+        } else {
+            debug("Features were listed for the plugin. Skipping server.xml.");
+            return new HashSet<String>();
+        }
+    }
+
+    /**
+     * Get the set of features defined in the server.xml
+     * @param serverDirectory The server directory containing the server.xml
+     * @return the set of features that should be installed from server.xml, or empty set if nothing should be installed
+     */
+    public Set<String> getServerFeatures(File serverDirectory) {
+        Set<String> result = getConfigDropinsFeatures(null, serverDirectory, "defaults");
+        result = getServerXmlFeatures(result, new File(serverDirectory, "server.xml"), null);
+        // add the overrides at the end since they should not be replaced by any previous content
+        return getConfigDropinsFeatures(result, serverDirectory, "overrides");
+    }
+    
+    /**
+     * Gets features from the configDropins's defaults or overrides directory
+     * 
+     * @param origResult
+     *            The features that have been parsed so far.
+     * @param serverDirectory
+     *            The server directory
+     * @param folderName
+     *            The folder under configDropins: either "defaults" or
+     *            "overrides"
+     * @return The set of features to install, or empty set if the cumulatively
+     *         parsed xml files only have featureManager sections but no
+     *         features to install, or null if there are no valid xml files or
+     *         they have no featureManager section
+     */
+    private Set<String> getConfigDropinsFeatures(Set<String> origResult, File serverDirectory, String folderName) {
+        Set<String> result = origResult;
+        File configDropinsFolder;
+        try {
+            configDropinsFolder = new File(new File(serverDirectory, "configDropins"), folderName).getCanonicalFile();
+        } catch (IOException e) {
+            // skip this directory if its path cannot be queried
+            warn("The " + serverDirectory + "/configDropins/" + folderName + " directory cannot be accessed. Skipping its server features.");
+            debug(e);
+            return result;
+        }
+        File[] configDropinsXmls = configDropinsFolder.listFiles(new FilenameFilter() {
+            @Override
+            public boolean accept(File dir, String name) {
+                return name.endsWith(".xml");
+            }
+        });
+        if (configDropinsXmls == null || configDropinsXmls.length == 0) {
+            return result;
+        }
+        // sort the files in alphabetical order so that overrides will happen in the proper order
+        Comparator<File> comparator = new Comparator<File>() {
+            @Override
+            public int compare(File left, File right) {
+                return left.getAbsolutePath().compareTo(right.getAbsolutePath());
```

Comment:
Also I think you should be using `getCanonicalPath` here as well instead of `getAbsolutePath`.

Reply:
configDropinsFolder is already canonical, so the list of files is just the file names under that canonical path.  In this case, we should just compare the file paths directly, and not resolve symbolic links.  Verified this with installUtility as well.